### PR TITLE
Simplifies registering metadata handlers.

### LINKED ISSUE #517: get rid of Dor::MetadataService.register
This is never used outside of dor-services and it makes it harder to read the code.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #517

### CODE REVIEW

File: lib/dor/services/metadata_service.rb [hunk 1]
```diff
@@ -69,13 +70,21 @@ def handler_for(prefix)
...
       private
 
+      # def handlers
+      #   @handlers ||= {}
+      # end
       def handlers
-        @handlers ||= {}
+        @handlers ||= {}.tap do |handlers|
+          # There's only one. If additional handlers are added, will need to be registered here.
+          register(CatalogHandler.new)
```
Comment (jcoyne) [inline lib/dor/services/metadata_service.rb]: I'd even get rid of the `register` logic and just return the `CatalogHandler` instance here.

File: lib/dor/services/metadata_service.rb [hunk 2]
```diff
@@ -69,13 +70,21 @@ def handler_for(prefix)
 
       private
 
+      # def handlers
+      #   @handlers ||= {}
+      # end
       def handlers
-        @handlers ||= {}
+        @handlers ||= {}.tap do |handlers|
```
Comment (jcoyne) [inline lib/dor/services/metadata_service.rb]: This inner variable is never used?

File: lib/dor/services/metadata_service.rb [hunk 3]
```diff
@@ -21,16 +22,16 @@ class MetadataService
...
-        %w(fetch label prefixes).each do |method|
-          raise TypeError, "Metadata handlers must define ##{method}" unless handler_class.instance_methods.include?(method) || handler_class.instance_methods.include?(method.to_sym)
-        end
-        handler = handler_class.new
-        handler.prefixes.each do |prefix|
-          handlers[prefix.to_sym] = handler
-        end
-        handler
-      end
+      # def register(handler_class)
```
Comment (jcoyne) [inline lib/dor/services/metadata_service.rb]: Can this stuff be removed?

File: lib/dor/services/metadata_service.rb (hunk 2 above)
Comment (jcoyne) [reply lib/dor/services/metadata_service.rb]: Would you mind calling the variable `handlers` something different than the method name?

File: lib/dor/services/metadata_service.rb [hunk 4]
```diff
@@ -62,20 +63,25 @@ def label_for(identifier)
...
 
       def handlers
-        @handlers ||= {}
+        @handlers ||= {}.tap do |handlers|
+          # There's only one. If additional handlers are added, will need to be registered here.
+          register(CatalogHandler.new, handlers)
+        end
+      end
+
+      def register(handler, handlers)
```
Comment (jcoyne) [inline lib/dor/services/metadata_service.rb]: would you mind renaming `handlers` so as not to confuse it with the method of the same name.

Comment (justinlittman) [reply lib/dor/services/metadata_service.rb]: So at least on line 73, rubocop insists:
```

File: lib/dor/services/metadata_service.rb [hunk 5]
```diff
@@ -62,20 +40,25 @@ def label_for(identifier)
...
+        raise MetadataError, "Unknown metadata prefix: #{prefix}" if handler.nil?
 
         handler
       end
 
       private
 
       def handlers
-        @handlers ||= {}
+        @handlers ||= {}.tap do |md_handlers|
```
Comment (jcoyne) [inline lib/dor/services/metadata_service.rb]: do we need `{}.tap ...` could this just be?
```ruby
@handlers ||= register(CatalogHandler.new, {})
```

Comment (jcoyne) [reply lib/dor/services/metadata_service.rb]: I guess the tap makes sense if the idea is to encourage more `register()` lines here.

**Pull request merged**